File: app-data/src/main/java/com/zxjdev/smile/data/authorization/datasource/AuthorizationCloudDataSource.java
package com.zxjdev.smile.data.authorization.datasource;

import javax.inject.Inject;

import io.reactivex.Observable;

public class AuthorizationCloudDataSource {

  private AuthorizationCloudService authorizationCloudService;

  @Inject
  public AuthorizationCloudDataSource(AuthorizationCloudService authorizationCloudService) {
    this.authorizationCloudService = authorizationCloudService;
  }

  public Observable<Void> register(String username, String password) {
    return authorizationCloudService.register(username, password);
  }

  public Observable<Void> login(String username, String password) {
    return authorizationCloudService.login(username, password);
  }

  public Observable<Boolean> checkIsLogined() {
    return authorizationCloudService.checkIsLogined();
  }

  public Observable<Void> logout() {
    return authorizationCloudService.logout();
  }
}

File: app-data/src/main/java/com/zxjdev/smile/data/authorization/datasource/AuthorizationCloudDataSource.kt
package com.zxjdev.smile.data.authorization.datasource

import javax.inject.Inject

class AuthorizationCloudDataSource @Inject constructor(
        private val authorizationCloudService: AuthorizationCloudService) {

    fun register(username: String, password: String) {
        authorizationCloudService.register(username, password)
    }

    fun login(username: String, password: String) {
        authorizationCloudService.login(username, password)
    }

    fun checkIsLogined(): Boolean {
        return authorizationCloudService.checkIsLogined()
    }

    fun logout() {
        authorizationCloudService.logout()
    }
}
